std-bits/base/src/main/java/org/enso/base/cache/ReloadDetector.java

```diff
@@ -0,0 +1,39 @@
+package org.enso.base.cache;
+
+import org.enso.base.polyglot.EnsoMeta;
+import org.graalvm.polyglot.Value;
+
+/**
+ * Detects that the reload button has been pressed.
+ *
+ * <p>.hasReloadOccurred() returns true if the reload button was pressed since the last call to
+ * .hasReloadOccurred().
+ *
+ * <p>This uses a weak reference (created in eval'd Enso code) that is set to null on reload.
+ */
+public class ReloadDetector {
+  // Weak reference that is set to null on reload.
+  private Value trigger;
```

Comment:
Is the comment still up-to-date? This contains a `Reload_Detector` now.

Reply:
Done.